raft/tracker/progress.go

```diff
@@ -55,9 +55,12 @@ type Progress struct {
 	// This is always true on the leader.
 	RecentActive bool
 
-	// ProbeSent is used while this follower is in StateProbe. When ProbeSent is
-	// true, raft should pause sending replication message to this peer until
-	// ProbeSent is reset. See ProbeAcked() and IsPaused().
+	// ProbeSent is true when a "probe" MsgApp was sent to this follower recently,
+	// and we haven't heard from it back yet. Used when the MsgApp flow is
+	// throttled, i.e. when State is StateProbe, or StateReplicate with saturated
+	// Inflights. In both cases, we need to continue sending MsgApp once in a
+	// while to guarantee progress, but we do so only when ProbeSent is false, to
+	// not overflow the receiver. See IsPaused().
```

Comment:
Might as well add here that there is periodicity here, i.e.

> but we only do so when ProbeSent is false (it is reset when receiving a heartbeat response) [...]

Reply:
Done.